src/components/workspace/collabEditing/helper.ts

```diff
@@ -1,40 +1,83 @@
-import { LINKS } from '../../../utils/constants';
-
-export function checkSessionIdExists(
-  sessionId: string,
-  onSessionIdExists: () => void,
-  onSessionIdNotExist: () => void,
-  onServerUnreachable: () => void
-) {
-  const xmlhttp = new XMLHttpRequest();
-  xmlhttp.onreadystatechange = () => {
-    if (xmlhttp.readyState === 4 && xmlhttp.status === 200) {
-      // Successfully reached server to verify ID
-      const state = JSON.parse(xmlhttp.responseText).state;
-      if (state === true) {
-        // Session ID exists
-        onSessionIdExists();
-      } else {
-        onSessionIdNotExist();
-      }
-    } else if (xmlhttp.readyState === 4 && xmlhttp.status !== 200) {
-      // Cannot reach server
-      onServerUnreachable();
-    }
-  };
-
-  xmlhttp.open('GET', 'https://' + LINKS.SHAREDB_SERVER + 'gists/' + sessionId, true);
-  xmlhttp.send();
-}
-
-export function createNewSession(onSessionCreated: (sessionId: string) => void) {
-  const xmlhttp = new XMLHttpRequest();
-  xmlhttp.onreadystatechange = () => {
-    if (xmlhttp.readyState === 4 && xmlhttp.status === 200) {
-      const id = JSON.parse(xmlhttp.responseText).id;
-      onSessionCreated(id);
-    }
-  };
-  xmlhttp.open('GET', 'https://' + LINKS.SHAREDB_SERVER + 'gists/latest/', true);
-  xmlhttp.send();
-}
+import { LINKS } from '../../../utils/constants';
+
+enum XMLHttpReadyState {
+  UNSENT = 0,
+  OPENED = 1,
+  HEADERS_RECEIVED = 2,
+  LOADING = 3,
+  DONE = 4
+}
+
+enum XMLHttpStatus {
+  OK = 200,
+  'Page not found' = 404
```

Comment:
Why is this breaking convention? Every other enum is in caps + snake_case. You can do `PAGE_NOT_FOUND`